packages/drivers/odsp-driver/src/createFile.ts

```diff
@@ -195,19 +197,48 @@ export async function createNewFluidFileFromSummary(
             logger,
             { eventName: "createNewFile" },
             async (event) => {
-                const { url, headers } = getUrlAndHeadersWithAuth(
-                    initialUrl, storageToken, forceAccessTokenViaAuthorizationHeader);
-                headers["Content-Type"] = "application/json";
+                const snapshotBody = JSON.stringify(containerSnapshot);
+                let url: string;
+                let headers: { [index: string]: string; };
+                let addInBody = false;
+                const formBoundary = uuid();
+                let postBody = `--${formBoundary}\r\n`;
+                postBody += `Authorization: Bearer ${storageToken}\r\n`;
+                postBody += `X-HTTP-Method-Override: POST\r\n`;
+                postBody += `Content-Type: application/json\r\n`;
+                postBody += `_post: 1\r\n`;
+                postBody += `\r\n${snapshotBody}\r\n`;
+                postBody += `\r\n--${formBoundary}--`;
+
+                if (postBody.length <= maxUmpPostBodySize) {
+                    const urlObj = new URL(initialUrl);
+                    urlObj.searchParams.set("ump", "1");
+                    url = urlObj.href;
+                    headers = {
+                        "Content-Type": `multipart/form-data;boundary=${formBoundary}`,
+                    };
+                    addInBody = true;
+                } else {
+                    const parts = getUrlAndHeadersWithAuth(
+                        initialUrl, storageToken, forceAccessTokenViaAuthorizationHeader);
+                    url = parts.url;
+                    headers = {
+                        ...parts.headers,
+                        "Content-Type": "application/json",
+                    };
+                    postBody = snapshotBody;
+                }
 
```

Comment:
I wonder if there is a chance to extract code that uses multi-part body into utility and use it across all places where we employ ump=1 optimization. Non-blocking

Reply:
Yes. I will follow up on this.